src/keyring/KeyringController.ts

```diff
@@ -577,6 +608,129 @@ export class KeyringController extends BaseController<
     this.update({ keyrings: [...keyrings] });
     return privates.get(this).keyring.fullUpdate();
   }
+
+  // QR Hardware related methods
+
+  /**
+   * Add qr hardware keyring.
+   *
+   * @returns The added keyring
+   */
+  private async addQRKeyring(): Promise<QRKeyring> {
+    const keyring = await privates
+      .get(this)
+      .keyring.addNewKeyring(KeyringTypes.qr);
+    await this.fullUpdate();
+    return keyring;
+  }
+
+  /**
+   * Get qr hardware keyring.
+   *
+   * @returns The added keyring
+   */
+  async getOrAddQRKeyring(): Promise<QRKeyring> {
+    const keyring = privates
+      .get(this)
+      .keyring.getKeyringsByType(KeyringTypes.qr)[0];
+    return keyring || (await this.addQRKeyring());
```

Comment:
@soralit why is necessary to add a keyring instead of raising an error here?  I'm not sure if @mcmire already addressed this method and I'm missing something

Reply:
I think @mcmire has a suggestion to this method before. 
The reason is that I want to make the method can always get the QR keyring instance so that we don't need to handle so many errors in the App layer.